mmseg/datasets/custom.py

```diff
@@ -85,6 +86,32 @@ def __init__(self,
         self.test_mode = test_mode
         self.ignore_index = ignore_index
         self.reduce_zero_label = reduce_zero_label
+        self.custom_classes = False
+
+        # overwrite classes with custom classes
+        if classes:
+            self.custom_classes = True
+            if not isinstance(classes, (tuple, list)):
+                raise ValueError(
+                    f'Unsupported type {type(classes)} of classes.')
+
+            if self.CLASSES:
+                self.old_id_to_new_id = {}
+                matched_classes = 0
+                for i, c in enumerate(self.CLASSES):
+                    if c not in classes:
+                        self.old_id_to_new_id[
+                            i + int(self.reduce_zero_label)] = -1 + int(
+                                self.reduce_zero_label)
+                    else:
+                        self.old_id_to_new_id[i + int(
+                            self.reduce_zero_label)] = classes.index(c) + int(
+                                self.reduce_zero_label)
+                        matched_classes += 1
+                if matched_classes < len(classes):
+                    raise ValueError('classes is not a subset of CLASSES.')
+
+            self.CLASSES = classes
```

Comment:
Could we do something like [this](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/datasets/custom.py#L231-L254)?